eva_submission/biosamples_submission.py

```diff
@@ -0,0 +1,471 @@
+        'Person First Name': 'FirstName',
+        'Person Last Name': 'LastName'
+    }
+
+    def __init__(self, metadata_spreadsheet):
+        super().__init__()
+        self.metadata_spreadsheet = metadata_spreadsheet
+        self.reader = EVAXLSReader(self.metadata_spreadsheet)
+
+    def map_metadata_to_bsd_data(self):
+        payloads = []
+        for sample_row in self.reader.samples:
+            bsd_sample_entry = {'characteristics': {}}
+            for key in sample_row:
+                if sample_row[key]:
+                    if key in self.sample_mapping:
+                        self.apply_mapping(bsd_sample_entry, key, sample_row[key])
+                    elif key in self.accepted_characteristics:
+                        # other field  maps to characteristics
+                        self.apply_mapping(
+                            bsd_sample_entry['characteristics'],
+                            self.map_sample_key(key.lower()),
+                            [{'text': sample_row[key]}]
+                        )
+                    else:
+                        # Ignore the other values
+                        pass
+            if sample_row.get('Novel attribute(s)'):
+                for novel_attribute in sample_row.get('Novel attribute(s)').split(','):
+                    attribute, value = novel_attribute.split(':')
+                    self.apply_mapping(
+                        bsd_sample_entry['characteristics'],
+                        self.map_sample_key(attribute.lower()),
+                        [{'text': value}]
+                    )
+
+            grouped_values = {}
+            project_row = self.reader.project
+            for key in project_row:
+                # Organisation and contact can contain multiple values that are split across several fields
+                # this will group the across fields
+                groupname = self.map_project_key(key.split()[0].lower())
+                if groupname in ['organization', 'contact']:
+                    self._group_across_fields(grouped_values, key, project_row[key])
+                elif groupname in self.project_mapping:
+                    # All the other project level field are added to characteristics
+                    self.apply_mapping(bsd_sample_entry['characteristics'], key.lower(),
+                                       [{'text': project_row[key]}])
+                else:
+                    # Ignore the other values
+                    pass
+            # Store the grouped values
+            for groupname in grouped_values:
+                self.apply_mapping(bsd_sample_entry, groupname, grouped_values[groupname])
+
+            bsd_sample_entry['release'] = _now
+            payloads.append(bsd_sample_entry)
+
+        return payloads
+
+    def submit_to_bioSamples(self):
+        sample_data = self.map_metadata_to_bsd_data()
+
+        # TODO: Only accessioned if it was not done before
+        # Check that the data
```

Comment:
Is further description missing here after "Check that the data"?

Reply:
That was refactored and additional comment added. Thanks